testnet/stacks-node/src/config.rs

```diff
@@ -1202,19 +1221,49 @@ impl FeeEstimationConfig {
         }
     }
 
-    pub fn make_scalar_fee_estimator<CM: CostMetric>(
+    pub fn make_scalar_fee_estimator<CM: 'static + CostMetric>(
         &self,
         mut chainstate_path: PathBuf,
         metric: CM,
-    ) -> ScalarFeeRateEstimator<CM> {
+    ) -> Box<dyn FeeEstimator> {
         if let Some(FeeEstimatorName::ScalarFeeRate) = self.fee_estimator.as_ref() {
             chainstate_path.push("fee_estimator_scalar_rate.sqlite");
-            ScalarFeeRateEstimator::open(&chainstate_path, metric)
-                .expect("Error opening fee estimator")
+            Box::new(
+                ScalarFeeRateEstimator::open(&chainstate_path, metric)
+                    .expect("Error opening fee estimator"),
+            )
         } else {
             panic!("BUG: Expected to configure a scalar fee estimator");
         }
     }
+
+    // Creates a fuzzed WeightedMedianFeeRateEstimator with window_size 5. The fuzz
+    // is uniform with bounds [+/- 0.5].
+    pub fn make_fuzzed_weighted_median_fee_estimator<CM: 'static + CostMetric>(
+        &self,
+        mut chainstate_path: PathBuf,
+        metric: CM,
+    ) -> Box<dyn FeeEstimator> {
+        if let Some(FeeEstimatorName::FuzzedWeightedMedianFeeRate) = self.fee_estimator.as_ref() {
+            chainstate_path.push("fee_fuzzed_weighted_median.sqlite");
+            let underlying_estimator = WeightedMedianFeeRateEstimator::open(
+                &chainstate_path,
+                metric,
+                self.fee_rate_window_size
+                    .expect("Must set `fee_rate_window_size`.")
```

Comment:
I put an `expect` in here.. but where would the defaults go?

Reply:
The most straight-forward way to handle the defaults is probably to change the `FeeEstimationConfig` struct fields back to non-optional, and then update the `impl From<FeeEstimationConfigFile> for FeeEstimationConfig` method (lines 1166 and 1167) to something like:

```rust
            fee_rate_fuzzer_fraction: f.fee_rate_fuzzer_fraction.unwrap_or(0.1f64),
            fee_rate_window_size: f.fee_rate_window_size.unwrap_or(5),
```

That way, if the config _file_ doesn't define those values, the parsed config struct uses the defaults.

Reply:
fixed